src/mlpack/methods/reinforcement_learning/environment/frozen_lake.hpp

```diff
@@ -0,0 +1,414 @@
+/**
+ * @file methods/reinforcement_learning/environment/frozen_lake.hpp
+ * @author Alex Nguyen
+ *
+ * This file is an implementation of Frozen Lake task:
+ * https://gym.openai.com/envs/FrozenLake-v0/
+ *
+ * TODO: provide an option to use dynamics directly from OpenAI gym.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_ENVIRONMENT_FROZEN_LAKE_HPP
+#define MLPACK_METHODS_RL_ENVIRONMENT_FROZEN_LAKE_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <stack>
+
+namespace mlpack {
+namespace rl {
+
+/**
+ * Implementation of Cart Pole task.
+ */
+class FrozenLake
+{
+ public:
+  class State
+  {
+   public:
+    State()
+    { 
+      // Nothing to do here.
+    }
+
+    // Initialize a new state with current row and column at 0.
+    State(size_t nRows, size_t nCols) : 
```

Comment:
Can you comment on each parameter, see https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/reinforcement_learning/environment/cart_pole.hpp for an example. Also, might be good to use a more descriptive parameter description.